### DISCUSSION TITLE
Load multiple csv files

### DISCUSSION THREAD

**Discussion by maheswarGR** [grafana/grafana-csv-datasource#147] [Q&A] [answered]
Hi, 
Is there any way to load mutiple csv files by ; or, separation or all csv files from folder.. 
Can we specify multiple csv file paths in CSV datasource?

**Comment by marcusolsson** [comment, marked as answer]
Unfortunately, right now each data source can only query on CSV file.

**Reply by maheswarGR** [reply]
Each data source can only query on only one CSV file? @marcusolsson

**Reply by marcusolsson** [reply]
Yes. Can we please keep the discussion here, rather than opening several issues and discussions for it?

**Reply by maheswarGR** [reply]
yes ..thank you for ur reply ..sorry for another issue created..

**Reply by maheswarGR** [reply]
is there any other data source which imports multiple csv files in grafana?

**Reply by mikethebee** [reply]
Are you able to merge the csv's into one and split using queries? I found the csv plugin really useful for simple input that saves time building a db schema that is the usual dashboarding approach.

**Reply by maheswarGR** [reply]
split using queries? queries in csv ? can u explain it bit more please

actually am in idea of importing
1. a folder path where there are many csv files 
                or 
2. specifying all paths of csv files separated by , or ;
is that also not possible ?

**Reply by mikethebee** [reply]
That is not a feature as was stated previously. Data normally needs pre-processing to create a single csv, or to use the database. I use scripts to select my data and format into a single csv, or with different data I use multiple data-sources linked to each appropriate csv.

You can merge most simply by concatenation using "type *.csv >new.csv" in windows command window, or "cat *.csv >new.csv" in Linux. For using a single new.csv in the csv plugin all files merged must have the same number of fields. Hopefully that helps clarify my suggestion.

**Reply by maheswarGR** [reply]
yes ..I filter data using transforms in csv ..that works ..
Got clarification ..
thank you for your humble reply ..
